packages/core/src/tracing/instrumentation/messaging/kafkaJs.js

```diff
@@ -313,22 +334,25 @@ function instrumentedEachBatch(originalEachBatch) {
     }
 
     let traceId;
+    let longTraceId;
     let parentSpanId;
     let level;
     if (batch.messages) {
       // Look for the the newer string header format first.
       for (let msgIdx = 0; msgIdx < batch.messages.length; msgIdx++) {
-        if (batch.messages[msgIdx].headers && batch.messages[msgIdx].headers[constants.kafkaTraceIdHeaderNameString]) {
-          traceId = String(batch.messages[msgIdx].headers[constants.kafkaTraceIdHeaderNameString]);
+        if (batch.messages[msgIdx].headers && batch.messages[msgIdx].headers[constants.kafkaTraceIdHeaderName]) {
+          traceId = String(batch.messages[msgIdx].headers[constants.kafkaTraceIdHeaderName]);
+          if (traceId) {
+            const limited = limitTraceId({ traceId });
```

Comment:
Note: Limiting the incoming trace ID here to 16 characters is one crucial change of this PR.